src/backend/weaviate/weaviate_manager.py

```diff
@@ -64,4 +83,180 @@ def add_weaviate_customer_class(self,customer_guid):
+                        {
+                            "name": "text",
+                            "dataType": ["text"],
+                            "description": "The chunked text content from the document.",
+                            "indexInverted": True
+                        },
+                        {
+                            "name": "chunk_number",
+                            "dataType": ["int"],
+                            "description": "An object containing metadata like chunk number",
+                            "indexInverted": False
+                        },
+                        {
+                            "name": "page_numbers",
+                            "dataType": ["int[]"],
+                            "description": "An object containing metadata like page number",
+                            "indexInverted": False
+                        },
+                        {
+                            "name": "customer_guid",
+                            "dataType": ["text"],
+                            "description": "A unique identifier for the customer (namespace-like isolation).",
+                            "indexInverted": True
+                        },
+                        {
+                            "name": "filename",
+                            "dataType": ["text"],
+                            "description": "The name of the file this chunk originates from.",
+                            "indexInverted": True
+                        }
+                    ]
+                }
+                self.client.schema.create_class(schema_obj)
+                logger.info(f"Schema '{class_name}' created successfully!")
+            else:
+                logger.info(f"Schema '{class_name}' already exists, skipping creation.")
+
+        except Exception as e:
+            logger.error(f"Unexpected error creating schema: {e}")
+
+    def insert_data(self, customer_guid: str, file_path: str):
+        try:
+            # Ensure the class schema is created before inserting data
+            class_names = self.generate_weaviate_class_name(customer_guid)
+            self.create_schema_per_customer_guid(customer_guid)
+
+            # Remove embeddings for the file and start fresh embedding
+            filename = os.path.basename(file_path)
+            self.delete_objects_by_customer_and_filename(customer_guid, filename)
+
+            # Read and validate data
+            with open(file_path, "r", encoding="utf-8") as file:
+                data = json.load(file)
+
+            if not isinstance(data, list):
+                raise ValueError("Invalid JSON format: Expected a list of objects.")
+
+            # Process all embeddings first
+            texts = [entry["text"].strip() for entry in data if "text" in entry and "metadata" in entry]
+            embeddings = self.model.encode(texts).tolist()
+
+            for index, (entry, embedding) in enumerate(zip(data, embeddings)):
```

Comment:
We were doing batch object creation in our earlier PR
Looks like that code is lost or changed

Whenever you are sending code changes, dont blindly update whatever code AI is giving

Check which parts of the code should be changed and then change it


Otherwise we will go into these unnecessary loops